crates/transaction-pool/src/identifier.rs

```diff
@@ -101,13 +101,15 @@ impl TransactionId {
     /// This returns `transaction_nonce - 1` if `transaction_nonce` is higher than the
     /// `on_chain_nonce`
     pub fn ancestor(transaction_nonce: u64, on_chain_nonce: u64, sender: SenderId) -> Option<Self> {
+        // SAFETY: transaction_nonce > on_chain_nonce ⇒ transaction_nonce >= 1
         (transaction_nonce > on_chain_nonce)
-            .then(|| Self::new(sender, transaction_nonce.saturating_sub(1)))
+            .then(|| Self::new(sender, unsafe { transaction_nonce.unchecked_sub(1) }))
     }
 
     /// Returns the [`TransactionId`] that would come before this transaction.
     pub fn unchecked_ancestor(&self) -> Option<Self> {
-        (self.nonce != 0).then(|| Self::new(self.sender, self.nonce - 1))
+        // SAFETY: self.nonce != 0 ⇒ self.nonce >= 1
+        (self.nonce != 0).then(|| Self::new(self.sender, unsafe {self.nonce.unchecked_sub(1)}))
```

Comment:
hmm, this is effectively already the same as `- 1` so we can keep it as is but can add the SAFTY note for clarity